kedro/framework/cli/starters.py

```diff
@@ -62,6 +142,69 @@ def _remove_readonly(func: Callable, path: Path, excinfo: Tuple):  # pragma: no
     func(path)
 
 
+def _is_valid_starter_entrypoint_config(
+    module_name: str, config: KedroStarterSpec
+) -> bool:
+    if not isinstance(config, KedroStarterSpec):
+        warn(
+            f"The starter configuration loaded from module {module_name}"
+            f"should be a 'KedroStarterSpec', got '{type(config)}' instead"
+        )
+        return False
+    return True
+
+
+def _get_starters_aliases() -> List[KedroStarterSpec]:
+    """This functions lists all the starters aliases declared in
+    the core repo and in plugins entry points.
+    The output looks like:
+    [
+        KedroStarterSpec(
+            name="astro-airflow-iris",
+            template_path="git+https://github.com/kedro-org/kedro-starters.git",
+            directory=None,
+            origin="kedro",
+        ),
+        KedroStarterSpec(
+            name="astro-iris",
+            template_path="git+https://github.com/kedro-org/kedro-starters.git",
+            directory="astro-airflow-iris",
+            origin="kedro",
+        ),
+    ]
+    """
+    # add an extra key to indicate from where the plugin come from
+    starters_aliases = _OFFICIAL_STARTERS_SPECS
+
+    existing_names: Dict[str, str] = {}  # dict {name: module_name}
+    for starter_entry_point in importlib_metadata.entry_points().select(
```

Comment:
Is there any chance of just calling the new entrypoint function that you modified recently or is this sufficiently different that we need to copy and paste the right bits here? (Not a big problem if we do, just wondering if your new scheme might make it easier to be more DRY now)

Reply:
I thought about this too! And I roughly remember it wasn't straightforward, will revisit it.